test/tests.ts

```diff
@@ -448,7 +447,48 @@ describe('Specifying dockerfiles', () => {
 		expect(content.trim()).to.equal('correct');
 	});
 
-	it('should allow an arch specific dockerfile to be specified', async () => {
+	it('should allow a Dockerfile.template to have a different name', async () => {
+		const stream = fs.createReadStream(
+			require.resolve('./test-files/SpecifiedMyDockerfileTemplate/archive.tar'),
+		);
+
+		let content: string;
+		const resolverName = 'Dockerfile.template';
+		const dockerfileName = 'MyDockerfile';
+
+		const hook = hookContent => {
+			content = hookContent;
+		};
+
+		const bundle = new Resolve.Bundle(stream, '', '', hook);
+		const [resolvePromise, listeners] = getPromiseForEvents(
+			{
+				resolver: (rsvrName: string) => expect(rsvrName).to.equal(resolverName),
+				'resolved-name': (resolvedName: string) =>
+					expect(resolvedName).to.equal(dockerfileName),
```

Comment:
You may want to ensure these `expect`s are actually called by assigning to a local variable and asserting they're `true` just after the `await` below. (I'm not completely sure this will work with `async/await` though, you may need to wrap them in an object.)

Reply:
Indeed. There were many other (pre-existing) tests doing the same thing, and again a situation of "90% code duplication". So I created a function and changed all tests to call it. Now even with the additional tests and comments, the PR has an overall net reduction of ~100 lines of code.